src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -323,35 +329,149 @@ FancyZones::VirtualDesktopInitialize() noexcept
+                int left = activeMi.rcWork.left + leftScale * activeW;
+                int top = activeMi.rcWork.top + topScale * activeH;
+
+                int width = windowRect.right - windowRect.left;
+                int height = windowRect.bottom - windowRect.top;
+
+                RECT newPosition{ left,
+                                  top,
+                                  min(left + width, activeMi.rcWork.right),
+                                  min(top + height, activeMi.rcWork.bottom) };
+
+                SizeWindowToRect(window, newPosition);
+            }
+        }
+    }
+}
+
+void FancyZones::MoveWindowIntoZone(HWND window, winrt::com_ptr<IZoneWindow> zoneWindow, const std::vector<int>& zoneIndexSet) noexcept
+{
+    auto& fancyZonesData = JSONHelpers::FancyZonesDataInstance();
+    if (!fancyZonesData.IsAnotherWindowOfApplicationInstanceZoned(window, zoneWindow->UniqueId()))
+    {
+        m_windowMoveHandler.MoveWindowIntoZoneByIndexSet(window, zoneIndexSet, zoneWindow);
+        fancyZonesData.UpdateProcessIdToHandleMap(window, zoneWindow->UniqueId());
+    }
+}
+
 // IFancyZonesCallback
 IFACEMETHODIMP_(void)
 FancyZones::WindowCreated(HWND window) noexcept
 {
     std::shared_lock readLock(m_lock);
-
-    if (m_settings->GetSettings()->appLastZone_moveWindows && IsInterestingWindow(window, m_settings->GetSettings()->excludedAppsArray))
+    if (m_settings->GetSettings()->appLastZone_moveWindows && ShouldProcessNewWindow(window))
     {
-        auto zoneWindow = m_workAreaHandler.GetWorkArea(window);
-        if (zoneWindow)
+        HMONITOR primary = MonitorFromWindow(nullptr, MONITOR_DEFAULTTOPRIMARY);
+        HMONITOR active = primary;
+
+        POINT cursorPosition{};
+        if (GetCursorPos(&cursorPosition))
         {
-            const auto activeZoneSet = zoneWindow->ActiveZoneSet();
-            if (activeZoneSet)
-            {
-                auto& fancyZonesData = JSONHelpers::FancyZonesDataInstance();
+            active = MonitorFromPoint(cursorPosition, MONITOR_DEFAULTTOPRIMARY);
+        }
 
-                wil::unique_cotaskmem_string guidString;
-                if (SUCCEEDED(StringFromCLSID(activeZoneSet->Id(), &guidString)))
-                {
-                    std::vector<int> zoneIndexSet = fancyZonesData.GetAppLastZoneIndexSet(window, zoneWindow->UniqueId(), guidString.get());
-                    if (zoneIndexSet.size() &&
-                        !IsSplashScreen(window) &&
-                        !fancyZonesData.IsAnotherWindowOfApplicationInstanceZoned(window, zoneWindow->UniqueId()))
-                    {
-                        m_windowMoveHandler.MoveWindowIntoZoneByIndexSet(window, zoneIndexSet, zoneWindow);
-                        fancyZonesData.UpdateProcessIdToHandleMap(window, zoneWindow->UniqueId());
-                    }
-                }
-            }
+        bool primaryActive = (primary == active);

```

Comment:
What are the chances that `GetCursorPosition` will fail and primaryActive is false? This is just fail-safe mechanism,right?

Reply:
fail safe mechanism is that I assign `HMONITOR active = primary` before doing `GetCursorPosition`
`primaryActive` will in that case be true, in all other scenarios it will have correct value